cpp/src/Ice/PropertyUtil.cpp

```diff
@@ -53,3 +56,38 @@ IceInternal::findProperty(string_view key, const PropertyArray* propertyArray)
 
     return nullopt;
 }
+
+void
+IceInternal::validatePropertyPrefix(
+    std::string_view propertyClass,
+    string_view prefix,
+    const PropertiesPtr properties,
+    const PropertyArray* propertyArray)
+{
+    vector<string> unknownProps;
+    PropertyDict props = properties->getPropertiesForPrefix(string{prefix} + ".");
+
+    // Do not check for unknown properties if Ice prefix, ie Ice, Glacier2, etc
+    for (const char** i = IceInternal::PropertyNames::clPropNames; *i; ++i)
+    {
+        string icePrefix = string(*i) + ".";
+        if (prefix.find(icePrefix) == 0)
+        {
+            return;
+        }
+    }
+
+    for (const auto& p : props)
+    {
+        // Plus one to include the dot.
+        if (!findProperty(p.first.substr(prefix.size() + 1), propertyArray))
+        {
+            unknownProps.push_back(p.first);
+        }
+    }
+
+    if (!unknownProps.empty())
+    {
+        throw UnknownPropertyException(__FILE__, __LINE__, propertyClass, prefix, unknownProps);
```

Comment:
I think it would make much more sense to compute the message here, and simplify UnknownPropertyException's constructor. It only needs a message string.

Presumably, this is the only spot where we want to throw UnknownPropertyException.

Reply:
We currently throw it in 3 places. Once we consolidate the property validate more we could move it into this helper function.